src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -868,7 +928,15 @@ private static TimeSpan GetElapsedTime(long startingTimestamp) =>
     /// <summary>Provides information about the invocation of a function call.</summary>
     public sealed class FunctionInvocationResult
     {
-        internal FunctionInvocationResult(bool shouldTerminate, FunctionInvocationStatus status, FunctionCallContent callContent, object? result, Exception? exception)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FunctionInvocationResult"/> class.
+        /// </summary>
+        /// <param name="shouldTerminate">Indicates whether the caller should terminate the processing loop.</param>
+        /// <param name="status">Indicates the status of the function invocation.</param>
+        /// <param name="callContent">Contains information about the function call.</param>
+        /// <param name="result">The result of the function call.</param>
+        /// <param name="exception">The exception thrown by the function call, if any.</param>
+        public FunctionInvocationResult(bool shouldTerminate, FunctionInvocationStatus status, FunctionCallContent callContent, object? result, Exception? exception)
```

Comment:
Why does this need to be public? I don't want to expose this ctor. Something else needs to construct these? If so, I'd rather just have the default parameterless ctor and make all of the properties settable.

Reply:
When overriding the `ProcessFunctionCallAsync` I need a way to return a `FunctionInvocationResult`, can be a factory, static `Create`, any other option.

Reply:
Reverted with latest changes.